server/src/main/java/io/crate/metadata/RelationName.java

```diff
@@ -72,6 +72,13 @@ public static RelationName ofQualified(String defaultSchema, QualifiedName name)
 
     public static RelationName of(@Nullable String schema, String name) {
         assert name != null : "table name must not be null";
+        if (schema != null && containsIllegalCharacters(schema)) {
+            throw new InvalidSchemaNameException(schema);
+        }
+        if (containsIllegalCharacters(name)) {
+            var invalidRelationName = new RelationName(schema, name);
+            throw new InvalidRelationName(invalidRelationName);
+        }
```

Comment:
Thanks for reviewing, @mfussenegger.
I needed the refactoring in order to construct the `RelationName` and raise `InvalidRelationName` exception.

Reply:
It would also be possible to change the `InvalidRelationName` exception to not implement `TableScopeException` (because it's not for a resolved table) and have a version that takes schema + name instead of relationName. But this looks good as well

Reply:
I will go with your suggestion as it makes more sense and to prevent the big refactoring.